src/components/Session/withAuthentication.js

```diff
@@ -13,7 +13,19 @@ const withAuthentication = Component => {
         authUser: null,
       };
     }
+    
+    componentDidMount() {
+      this.listener = this.props.firebase.auth.onAuthStateChanged(
+        this.setState({authUser: null})
+      );
+    }
 
+    componentWillUnmount() {
+      this.listener = this.props.firebase.auth.onAuthStateChanged(
```

Comment:
Instead of setting new listener. You should reset the listener by calling it. Something like
```
if (this.listener) {
  this.listener();
  this.listener = null;
}
```

Reply:
https://firebase.google.com/docs/reference/js/firebase.auth.Auth#onauthstatechanged